src/java_bytecode/java_class_loader.cpp

```diff
@@ -33,17 +33,24 @@ java_bytecode_parse_treet &java_class_loadert::operator()(
 {
   std::stack<irep_idt> queue;
 
-  // Always require java.lang.Object, as it is the base of
-  // internal classes such as array types.
-  queue.push("java.lang.Object");
-  // java.lang.String
-  queue.push("java.lang.String");
-  // add java.lang.Class
-  queue.push("java.lang.Class");
+  queue.push(class_name);
```

Comment:
Why is this reordering necessary now but not before?

Reply:
We want java.lang.Object to be the first thing loaded. Otherwise, we may get objects that inherited from java_root_class (since there is thus far no loaded java.lang.Object) and then were referenced to java.lang.Object (thus the sizes may not match if the model has additional fields).